plugins/languages/treesitter/treesitter.nix

```
@@ -12,6 +12,101 @@ helpers.neovim-plugin.mkNeovimPlugin config {
   luaName = "nvim-treesitter.configs";
   defaultPackage = pkgs.vimPlugins.nvim-treesitter;
 
+  description = ''
+    Provides an interface to [tree-sitter]
+
+    ## Installing Your Own Grammars with Nixvim
+
+    The `nvim-treesitter` derivation generates the grammar packages based on the
+    default configs of nvim-treesitter but you can always install your own
+    grammars.
+
+    ```nix
+    { pkgs, ... }:
+    let
+      #example of adding your own grammar:
+      treesitter-nu-grammar = pkgs.tree-sitter.buildGrammar {
+        language = "nu";
+        version = "0.0.0+rev=0bb9a60";
+        src = pkgs.fetchFromGitHub {
+          owner = "nushell";
+          repo = "tree-sitter-nu";
+          rev = "0bb9a602d9bc94b66fab96ce51d46a5a227ab76c";
+          hash = "sha256-A5GiOpITOv3H0wytCv6t43buQ8IzxEXrk3gTlOrO0K0=";
+        };
+        meta.homepage = "https://github.com/nushell/tree-sitter-nu";
+      };
+
+      # or you can yoink any grammars in tree-sitter.grammars.$\{grammar-name}
+      # treesitter-nu-grammar = pkgs.tree-sitter-grammars.tree-sitter-nu;
+    in
+    {
+
+      programs.nixvim.plugins = {
+        treesitter = {
+          enable = true;
+          indent = true;
+          grammarPackages = pkgs.vimPlugins.nvim-treesitter.passthru.allGrammars ++ [
+            treesitter-nu-grammar
+          ];
+        };
+      };
+
+      programs.nixvim.extraConfigLua =
+        '''
+          do
+            local parser_config = require "nvim-treesitter.parsers".get_parser_configs()
```

Nit: I find examples are easier to read with explicit parens on function calls.
```suggestion
            local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
```

(I haven't checked if the same change is needed elsewhere in the example)

---

This was the only instance that I could find.